File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/1558.py
import math

while True:
    try:
        n = int(input())
        
        ini = 0
        existe = False
        
        if(n >= 0):
            fim = int(math.sqrt(n))
            
            while fim >= ini:
                temp = ini**2 + fim**2
                
                if temp == n:
                    existe = True
                    break
                elif temp < n:
                    ini += 1
                else:
                    fim -= 1
           
        if existe: print("YES")
        else: print("NO")
        
    except:
        break

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/1558.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {
    int n;
    
    while(cin >> n){
        
        int ini = 0, fim = sqrt(n);
        bool existe = false;
        
        while(fim >= ini){
            int temp = ini * ini + fim * fim;
            
            if(temp == n){
                existe = true;
                break;    
            } else if(temp < n)
                ini++;
            else
                fim--;
            
        }
        
        if(existe) cout << "YES" << endl;
        else cout << "NO" << endl;
        
    }
    
    return 0;
}